bundles/org.openhab.binding.smarther/src/main/java/org/openhab/binding/smarther/internal/discovery/SmartherModuleDiscoveryService.java

```diff
@@ -0,0 +1,172 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.smarther.internal.discovery;
+
+import static org.openhab.binding.smarther.internal.SmartherBindingConstants.*;
+
+import java.lang.invoke.MethodHandles;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.DiscoveryService;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerService;
+import org.openhab.binding.smarther.internal.account.SmartherAccountHandler;
+import org.openhab.binding.smarther.internal.api.dto.Location;
+import org.openhab.binding.smarther.internal.api.dto.Module;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@code SmartherModuleDiscoveryService} queries the Smarther API gateway to discover available Chronothermostat
+ * modules inside existing plants registered under the configured Bridges.
+ *
+ * @author Fabio Possieri - Initial contribution
+ */
+@Component(service = DiscoveryService.class, immediate = true, configurationPid = "discovery.smarther")
+@NonNullByDefault
+public class SmartherModuleDiscoveryService extends AbstractDiscoveryService implements ThingHandlerService {
```

Comment:
ThingHandlerService instances should not be osgi components. Osgi components are singletons whereas a new ThingHandlerService instance is created for each handler that uses it as a services. So pick one or the other.

Reply:
Picked up the `ThingHandlerService` instance option, thanks